core/src/main/java/tech/pegasys/web3signer/core/service/http/handlers/keymanager/delete/KeystoreFileManager.java

```diff
@@ -55,6 +59,37 @@ public void deleteKeystoreFiles(final String pubkey) throws IOException {
     }
   }
 
+  /**
+   * Create Keystore metadata, json and password files
+   *
+   * @param fileNameWithoutExtension File name, usually public key, without extension.
+   * @param jsonKeystoreData Keystore Json data which will be written to
+   *     fileNameWithoutExtension.json
+   * @param password password that will be written as fileNameWithoutExtension.password.
+   * @throws IOException In case file write operations fail
+   */
+  public void createKeystoreFiles(
+      final String fileNameWithoutExtension, final String jsonKeystoreData, final String password)
+      throws IOException {
+    final Path metadataYamlFile =
+        keystorePath.resolve(fileNameWithoutExtension + METADATA_YAML_EXTENSION);
+    final Path keystoreJsonFile =
+        keystorePath.resolve(fileNameWithoutExtension + KEYSTORE_JSON_EXTENSION);
+    final Path keystorePasswordFile =
+        keystorePath.resolve(fileNameWithoutExtension + KEYSTORE_PASSWORD_EXTENSION);
+
+    // create yaml file first (so that if it fails we haven't written password file before it)
+    final FileKeyStoreMetadata data =
```

Comment:
Thinking the create should be atomic and transactional so if any individual file fails to create we clean up by deleting the partially created one. So basically thinking we should add a catch and do the deletes in this function.